pkg/serverless/serverless.go

```diff
@@ -268,6 +273,7 @@ func WaitForNextInvocation(stopCh chan struct{}, daemon *Daemon, metricsChan cha
 		daemon.ComputeGlobalTags(payload.InvokedFunctionArn, config.GetConfiguredTags(true))
 		aws.SetARN(payload.InvokedFunctionArn)
 		daemon.StartInvocation()
+		daemon.detectTimeout(payload.DeadlineMs, safetyBufferTimout, daemon.handleTimeout)
```

Comment:
How do we disable the timeout? A slightly more idiomatic way of handling this in go is to use a context object with timeout, https://golang.org/pkg/context/#WithTimeout

Reply:
I have created a channel referenced in d.timeoutChan to stop the timeout but it indeed may be easier with `withTimeout`